lib/credentials/shared_ini_file_credentials.js

```diff
@@ -158,23 +159,75 @@ AWS.SharedIniFileCredentials = AWS.util.inherit(AWS.Credentials, {
         return;
       }
 
-      this.accessKeyId = profile['aws_access_key_id'];
-      this.secretAccessKey = profile['aws_secret_access_key'];
-      this.sessionToken = profile['aws_session_token'];
+      if (profile['aws_access_key_id'] || profile['aws_secret_access_key']) {
+        this.accessKeyId = profile['aws_access_key_id'];
+        this.secretAccessKey = profile['aws_secret_access_key'];
+        this.sessionToken = profile['aws_session_token'];
+      } else if (profile['credential_process']) {
+        this.loadViaCredentialProcess(profile, function(err, data) {
+          if (err) {
+            callback(err);
+          } else {
+            self.expired = false;
+            self.accessKeyId = data.AccessKeyId;
+            self.secretAccessKey = data.SecretAccessKey;
+            self.sessionToken = data.SessionToken;
+            if (data.Expiration) {
+              self.expireTime = new Date(data.Expiration);
+            }
+            callback(null);
+          }
+        });
+      }
 
-      if (!this.accessKeyId || !this.secretAccessKey) {
+      if ((!this.accessKeyId || !this.secretAccessKey) && !profile['credential_process']) {
         throw AWS.util.error(
           new Error('Credentials not set for profile ' + this.profile),
           { code: 'SharedIniFileCredentialsProviderFailure' }
         );
       }
       this.expired = false;
-      callback(null);
+      if (this.accessKeyId || this.secretAccessKey) {
```

Comment:
Should this be `this.accessKeyId && this.secretAccessKey`

Reply:
I think the control flow affected in this PR can be:

```js
if (profile['aws_access_key_id'] || profile['aws_secret_access_key']) {
  // set props & callback(null)
} else if (profile['credential_process']) {
  // loadViaCredentialProcess
} else {
  throw AWS.util.error(
          new Error('Credentials not set for profile ' + this.profile),
          { code: 'SharedIniFileCredentialsProviderFailure' }
        );
}
```

It's a bit confusing the way the conditions are currently being checked in multiple places.